AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/AndroidAnnotationsEnvironment.java

```diff
@@ -76,7 +76,7 @@
 
 	GeneratedClassHolder getGeneratedClassHolder(Element element);
 
-	ProcessHolder.Classes getClasses();
+	ClassesHolder.Classes getClasses();
```

Comment:
Damn... right after mergin i came up with a question.... don't the changes to `AndroidAnnotationsEnvironment` break existing plugins if the use this method?!

Reply:
@WonderCsabo @smaugho I hope I'm wrong with this thougt. :/

Reply:
Yes, it will. However i think we can allow breaking changes for plugins, but not allow breaking changes for users. WDYT?

Reply:
But we should raise the minor version, and document the changes.

Reply:
@WonderCsabo it depends on how active the plugin devs are and if they update the plugins to be compatible once again. out-of-date updates mean breaking change for the users of those pluigns.

Reply:
Hmm, you are correct. I am afraid we cannot have this change without moving to 5.0. So we have to revert moving the class.

Reply:
@smaugho can you do that?

Reply:
we could revert the PR merge and @smaugho can open a new PR with the other changes.

Reply:
@WonderCsabo, @dodgex I guess it should be done 😓 ... Could we open a new branch for version 5 realease and keep these changes there?

Reply:
Yes, please!

Reply:
And can you please open another PR which reverts moving `Classes`?

Reply:
Yes @WonderCsabo , will try to do it today or tomorrow. Please if possible create this branch for version 5, in order to submit this one there (so I guess, I have to make a PR to develop which reverts the changes, and another one to the branch for version 5, which remakes it.

Reply:
For now just move back Classes. For version 5.0, we will just revert that PR.

Reply:
Hi @WonderCsabo , @dodgex .. sorry that still didn't make the change... had quite a hectic week. I see you're considering let a breaking change on the release, I'll wait for your decision on that. I don't think that is something uncommon, I'm used to see that in many libs around, where a specific version works with some specific version of other lib (or with a version range of the other lib). 

Also in more codes, the "Classes" are used like: `getClasses().SOME_OBJECT`, and that will not be broken, how this will break it is if the programmer assigns the classes object, like `Classes classes = pluginHolder.getClassess()` ...

Reply:
it will break if the programmer imported the `Classes` class. Please revert that move.